File: Models/OutputMode.cs
namespace WSNH.Models;

/// <summary>
/// What happens to the rewritten text once the AI call finishes.
/// </summary>
public enum OutputMode
{
    /// <summary>Replace the selected text in place (simulated Ctrl+V).</summary>
    AutoPaste,
    /// <summary>Show the result in a small popup window to copy from.</summary>
    Popup,
    /// <summary>Ask first: Paste &amp; Replace, Copy Only, or Cancel.</summary>
    Ask
}

File: mac/Sources/WSNH/Models/OutputMode.swift
import Foundation

/// What WSNH should do with the AI's rewritten text once it comes back.
enum OutputMode: String, Codable, CaseIterable, Identifiable {
    case autoPaste
    case popup
    case ask

    var id: String { rawValue }

    var displayName: String {
        switch self {
        case .autoPaste: return "Auto-paste in place"
        case .popup: return "Show popup to review"
        case .ask: return "Ask each time"
        }
    }
}
